solver/src/solver/http_solver.rs

```diff
@@ -93,15 +97,7 @@ impl HttpSolver {
         }
     }
 
-    // Solver api requires specifying token as strings. We use the address as a string for now.
-    // Later we could use a more meaningful name like the token symbol but we have to ensure
-    // uniqueness.
-    fn token_to_string(&self, token: &H160) -> String {
-        // Token names must start with a letter.
-        format!("t{:x}", token)
-    }
-
-    fn map_tokens_for_solver(&self, orders: &[Liquidity]) -> HashMap<String, H160> {
+    fn map_tokens_for_solver(&self, orders: &[Liquidity]) -> Vec<H160> {
```

Comment:
nit: Instead of collecting into a `HashSet` to then collect into a `Vec`, can't this just return a `HashSet` right away?